src/config/builder.rs

```diff
@@ -209,7 +209,9 @@ impl ConfigBuilder {
 
         #[cfg(feature = "enterprise")]
         {
-            self.enterprise = with.enterprise;
+            if let Some(enterprise) = with.enterprise {
```

Comment:
I think we should do something more akin to [api.merge](https://github.com/vectordotdev/vector/blob/master/src/config/api.rs#L44) than a simple override. The benefits being:
- Allows us to combine fields where it makes sense (e.g. `tags`) and error where it doesn't (e.g. `site`). Thus, keeping the user in the loop and forcing them to make conscious decisions. This should help avoid accidents. 
- Allows us to be more transparent and upfront with merging behavior with users. In the current scheme, the order in which configurations are combined matters in a non-obvious way and no indication is given that one enterprise config overrides another entirely.

Reply:
Agreed. A merge strategy is in keeping with other sections. It also allows for flexible code organisation; a company could choose to keep an API key in a protected file, with remaining settings configurable by other teams in a separate file.

Reply:
@leebenson what should happen if we merge those 2?

```toml
# first.toml
[enterprise]
enabled = false
api_key = "foo"
# second.toml
[enterprise]
enabled = true
api_key = "bar"
```

Or what if we merge those 2 considering we are not sure of the order?
```toml
# first.toml
[enterprise]
api_key = "foo"
# second.toml
[enterprise]
api_key = "bar"
```

Reply:
In both examples, the 2nd file would overwrite existing keys.

Files should be parsed alphabetically, so there's some expectation of ordering.